app/src/test/kotlin/fr/free/nrw/commons/category/CategoriesModelTest.kt

```diff
@@ -27,26 +30,39 @@ class CategoriesModelTest {
         MockitoAnnotations.initMocks(this)
     }
 
-    // Test Case for verifying that Categories search (MW api calls) are case-insensitive
+    // Test Case for verifying that Categories search (MW api calls)
     @Test
     fun searchAllFoundCaseTest() {
         val categoriesModel = CategoriesModel(categoryClient, mock(), mock())
 
         val expectedList = listOf("Test")
-        whenever(categoryClient.searchCategoriesForPrefix("tes", 25))
+        whenever(
+            categoryClient.searchCategoriesForPrefix(
+                ArgumentMatchers.anyString(),
+                ArgumentMatchers.anyInt(),
+                ArgumentMatchers.anyInt()
+            )
+        )
             .thenReturn(Single.just(expectedList))
 
         // Checking if both return "Test"
         val expectedItems = expectedList.map { CategoryItem(it, false) }
-        categoriesModel.searchAll("tes", emptyList(), emptyList())
+        var categoryTerm = "Test"
```

Comment:
Given the fact that the first title character is case insensitive in MediaWiki [[ref](https://www.mediawiki.org/wiki/Manual:Page_title)], I don't think this test would help us catch bugs that we want it to catch. A safe case would be to have a set of test cases with search queries like:
* `TEST`
* `test`
* `Test Category`

Reply:
The aim was to achieve the test coverage of the functions which modify the search term toLowerCase(), the ultimate aim is to verify that the terms are passed as it it which the modified test cases do handle and that's why I used "Test" with a capital 'T' to verify that if you see the next line verifies that the categoryClient's corresponding function is called with the exact search term as passed to the function. I don't think we need to add all combinations of such cases since the test case already handles that. You can verify it by modifying the corresponding method in the CategoryModel. You will see that the test case would fail for that